eng/DownloadSharedSource.ps1

```diff
@@ -2,29 +2,54 @@
 $ErrorActionPreference = 'Stop'
 $ProgressPreference = 'SilentlyContinue'
 
-function DownloadAll([string[]]$files, [string]$baseUrl, [string]$downloadPath)
+$repoRoot = Resolve-Path (Join-Path $PSScriptRoot '..')
+$clonedPath = Join-Path $repoRoot 'artifacts/azure-sdk-for-net-shared'
+$azCoreSharedPath = "sdk/core/Azure.Core/src/Shared/"
+$armCoreSharedPath = "sdk/resourcemanager/Azure.ResourceManager/src/Shared/"
+
+if(Test-Path $clonedPath)
+{
+    Remove-Item $clonedPath -Recurse -Force
+}
+
+Write-Host "git clone --no-checkout --filter=tree:0 https://github.com/Azure/azure-sdk-for-net.git $clonedPath"
+git clone --no-checkout --filter=tree:0 https://github.com/Azure/azure-sdk-for-net.git $clonedPath
+Push-Location $clonedPath
+Write-Host "git sparse-checkout init"
+git sparse-checkout init
+Write-Host "git sparse-checkout set --no-cone $azCoreSharedPath $armCoreSharedPath"
+git sparse-checkout set --no-cone $azCoreSharedPath $armCoreSharedPath
+Write-Host "git checkout"
+git checkout
+Pop-Location
+
+function CopyAll([string[]]$files, [string]$source, [string]$destination)
 {
     foreach ($file in $files)
     {
-        Write-Host "Downloading" $file
-        $text = (Invoke-WebRequest -Uri "$baseUrl/$file").Content
-        $text.Trim() | Out-File (Join-Path $downloadPath $file)
+        Write-Host "Copying $file to $destination"
+        Copy-Item (Join-Path $source $file) (Join-Path $destination $file)
```

Comment:
Are there any existing changes in the shared source? I think there were some EOF newlines missing so we should probably make sure we update them with this script.

Reply:
Out-File was appending a new line to the end of the piped in string

I've corrected the 2 files that didn't have trailing new lines in the azure-sdk-for-net repo:
https://github.com/Azure/azure-sdk-for-net/pull/36250